haystack/core/component/component.py

```diff
@@ -147,6 +147,12 @@ def __call__(cls, *args, **kwargs):
                 if run_signature.parameters[param].default != inspect.Parameter.empty:
                     socket_kwargs["default_value"] = run_signature.parameters[param].default
                 instance.__haystack_input__[param] = InputSocket(**socket_kwargs)
+
+        # Since a Component can't be used in multiple Pipelines at the same time
+        # we need to know if it's already owned by a Pipeline when adding it to one.
+        # We use this flag to check that.
+        instance.__haystack_pipeline_owned__ = False
```

Comment:
nit: I would use a name with a simpler semantic, like `__haystack_added_to_pipeline__`

Reply:
I was actually thinking of changing this to hold a reference to the `Pipeline`, it can be useful in later PRs. 🤔 
So something like `__haystack_owned_by_pipeline__`.

Reply:
Since we don't have any explicit concept of ownership in Haystack, it's the `owned` bit that bugs me 😅

Reply:
Makes sense, I'll use added then.